### PR TITLE
[Gen] Always free strdup strings as they're not held by native

### PR DESCRIPTION
**Pull request by Therzok**
This fixes a few leaks caused by gstrdup-ing by firing and forgetting.

https://gist.github.com/Therzok/822d377e69476d172f434aa79ef74aed

### CODE REVIEW

**Comment by Therzok** [comment]
cc @DavidKarlas @alanmcgovern

**Comment by Therzok** [comment]
Actually, some of these bindings look incorrect. The char stuff should take in a simple char, gtk_label_get should be using an out string, ParseArgs should be using a list.

Only the paint/draw/pango stuff seem to be generated correctly now (and I'm not sure why the former two are being picked up by this kind of marshaller since those are const gchar/char).

**Comment by Therzok** [comment]
Updated diff because the previous comment is now obsolete as the fixes have to the API have landed:
https://github.com/mono/gtk-sharp/commit/22952685e11ac884b7a1d8312f14684825eee407
https://github.com/mono/gtk-sharp/commit/f66a1c84e0d6782340844dcb4feaa3e3871219c7
https://github.com/mono/gtk-sharp/commit/8abc2bab4791f205b3512bb2076469775fd3a49b

File: generator/MarshalGen.cs [hunk 1]
```diff
@@ -60,5 +60,22 @@ public override string FromNative (string var)
...
+		}
+
+		public string AllocNative (string managed_var)
+		{
+			return CallByName (managed_var);
+		}
+
+		public string ReleaseNative (string native_var)
+		{
+			return $"GLib.Marshaller.Free ({native_var})";
```
Comment (knocte) [inline generator/MarshalGen.cs]: Does gtk-sharp depend already in a high enough version of mono to support interpolated strings?

Comment (Therzok) [inline generator/MarshalGen.cs]: Jenkins does have Mono 4.4. I think we should bump the required version in the configure file.

In any case, I don't like this patch, but there's not much without rewriting the generator typesystem, and I prefer to do that in master.

Comment (knocte) [inline generator/MarshalGen.cs]: >  I think we should bump the required version in the configure file.

Probably, but not only for this little patch. How about doing that after releasing a new gtk#2.x version? @meebey needs it for smuxi.

Comment (Therzok) [inline generator/MarshalGen.cs]: We already depend on .NET 4.6 at build time (not runtime) for the string marshalling optimizations.

Comment (decriptor) [inline generator/MarshalGen.cs]: Is there going to actually be a gtk-sharp 3.0 release? It seems like there has "going" to be a release for the last 3 years :(

**Pull request merged**